### DISCUSSION TITLE
What is the sensible behaviour here?

### DISCUSSION THREAD

**Discussion by omkar111111** [encode/django-rest-framework#8435] [General]
```
Note that because we want to be able to POST to this view from clients that won't have a CSRF token we need to mark the view as csrf_exempt. This isn't something that you'd normally want to do, and REST framework views actually use more sensible behavior than this, but it'll do for our purposes right now.
```
Source: https://www.django-rest-framework.org/tutorial/1-serialization/

**Comment by charanjit-singh** [comment]
Didn't get your point

**Reply by hashlash** [reply]
I think the part being asked is the **more sensible behavior** on the quoted docs.

> ..., and REST framework views actually use **more sensible behavior** than this, ...

Btw @omkar111111 , it seems this discussion is more suited to the [Q&A category](https://github.com/encode/django-rest-framework/discussions/categories/question-answer).

**Comment by hashlash** [comment]
From what I understand, the **more sensible behavior** here is allowing all requests to not have a CSRF token except for session-based requests. This is due to the session-based requests usually having access to the generated CSRF token, while cross-domain doesn't.

This behavior is documented on [AJAX, CSRF & CORS](https://www.django-rest-framework.org/topics/ajax-csrf-cors/):

> AJAX requests that are made within the same context as the API they are interacting with will typically use `SessionAuthentication`. This ensures that once a user has logged in, any AJAX requests made can be authenticated using the same session-based authentication that is used for the rest of the website.
> 
> AJAX requests that are made on a different site from the API they are communicating with will typically need to use a non-session-based authentication scheme, such as `TokenAuthentication`.

And the corresponding code:

- The CSRF exempt (on `rest_framework.views.APIView.as_view` function)

  https://github.com/encode/django-rest-framework/blob/86673a337a4fe8861c090b4532379b97e3921fef/rest_framework/views.py#L144

- The CSRF enforcement (on `rest_framework.authentication.SessionAuthentication.authentication` function)

  https://github.com/encode/django-rest-framework/blob/86673a337a4fe8861c090b4532379b97e3921fef/rest_framework/authentication.py#L130
